src/medley/core.cljc

```diff
@@ -357,6 +357,64 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn partition-after
+  "Returns a lazy sequence of partitions, splitting after `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [v (volatile! [])
+           split? (volatile! false)]
+       (fn
+         ([] (rf))
+         ([result]
+          (rf (if-some [v (not-empty @v)]
+                (unreduced (rf result v))
+                result)))
+         ([result x]
+          (if @split?
+            (let [result (rf result @v)]
+              (vreset! v (if (reduced? result) [] [x]))
```

Comment:
Why is `v` reset when the result is reduced?

Reply:
Because the "completing" arity still cares about `v` even when the result is reduced. Note that I asserted this behaviour as the final assertion of each test, which would fail if we don't reset `v` when the result is reduced.

Reply:
Thanks for the explanation.